document/core/exec/instructions.rst

```diff
@@ -2680,14 +2680,40 @@ Control Instructions
 :math:`\RETHROW~l`
 ..................
 
+1. Assert: due to :ref:`validation <valid-rethrow>`, the stack contans at least :math:`l+1` labels.
+
+2. Let :math:`L` be the :math:`l`-th label appearing on the stack, starting from the top and counting from zero.
+
+3. Assert: due to :ref:`validation <valid-rethrow>`, :math:`L` is a catch label, i.e., a label of the form :math:`(\LCATCH~[t^\ast])`.
+
+4. Repeat :math:`l+1` times:
+
+   a. While the instruction at the top of the stack is a value, a |handler|, or a |CAUGHTadm| instruction, do:
```

Comment:
```suggestion
   a. While the top of the stack is a value, a |handler|, or a |CAUGHTadm| instruction, do:
```
Nit: It's unclear we can call values instructions, and rules for `br` and `return` also just say "While the top of the stack ..."